drivers/dp/swdp_ll_pin.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+#define CPU_CLOCK		64000000U
+#else
+#define CPU_CLOCK		CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#endif
+
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF53X)
+#define FAST_BITBANG_HW_SUPPORT 1
+#else
+#define FAST_BITBANG_HW_SUPPORT 0
+#endif
```

Comment:
Having one header file to encapsulate all future low-level bitbang implementations likely wont scale?

Reply:
It is okay for now (82 lines), we will see how it evolves and react. But implementing faster GPIO for all platforms is not our goal here.